src/EditorFeatures/TestUtilities/ChangeSignature/AbstractChangeSignatureTests.cs

```diff
@@ -66,10 +75,29 @@ protected override ParseOptions GetScriptOptions()
             bool verifyNoDiagnostics = false,
             ParseOptions parseOptions = null,
             int expectedSelectedIndex = -1)
+            => await TestChangeSignatureViaCommandAsync(languageName, markup,
+                updatedSignature?.Select(i => new AddedParameterOrExistingIndex(i)).ToArray(),
+                expectedSuccess, expectedUpdatedInvocationDocumentCode,
+                expectedErrorText,
+                totalParameters,
+                verifyNoDiagnostics,
+                parseOptions,
+                expectedSelectedIndex);
+
+        internal async Task TestChangeSignatureViaCommandAsync(
+            string languageName,
+            string markup,
+            AddedParameterOrExistingIndex[] updatedSignature,
+            bool expectedSuccess = true,
+            string expectedUpdatedInvocationDocumentCode = null,
+            string expectedErrorText = null,
+            int? totalParameters = null,
+            bool verifyNoDiagnostics = false,
+            ParseOptions parseOptions = null,
+            int expectedSelectedIndex = -1)
```

Comment:
holy crap.  can we just make a SignatureChangeParameters object?  it can have read/write properties on it, allowing the caller to just easily instantiate and set the values they care about.

Reply:
Filed https://github.com/dotnet/roslyn/issues/42734 to clean this up.